src/mpi/misc/memory_alloc_kinds.c

```diff
@@ -50,13 +58,38 @@ int MPIR_get_supported_memory_kinds(char *requested_kinds, char **out_kinds)
 
 /*** static functions ***/
 
+/* return true if the input kind is an unsupported GPU kind */
+static bool is_unsupported_gpu(const char *kind)
+{
+    int gpu_type;
+    int gpu_supported;
+
+    if (strcmp(kind, "cuda") == 0) {
+        gpu_type = MPIX_GPU_SUPPORT_CUDA;
+    } else if (strcmp(kind, "rocm") == 0) {
+        gpu_type = MPIX_GPU_SUPPORT_HIP;
+    } else if (strcmp(kind, "level_zero") == 0) {
+        gpu_type = MPIX_GPU_SUPPORT_ZE;
+    } else {
+        return false;
+    }
+
+    (void) MPIR_GPU_query_support_impl(gpu_type, &gpu_supported);
+    return !gpu_supported;
+}
+
 static bool is_supported(const char *kind)
```

Comment:
Orthogonal to this PR, but I find it difficult to peruse the parsing logics. How about adding some example syntax in comments?

Reply:
Will do.

Reply:
Checked the added comment in the WIP commit and let me know if that's what you had in mind.

Reply:
Looks good!